File: src/org/sakura/settings/utils/SystemUtils.java
/*
 * SPDX-FileCopyrightText: 2025 Evolution X
 * SPDX-License-Identifier: Apache-2.0
 */

package org.sakura.settings.utils;

import android.app.ActivityManager;
import android.app.AlertDialog;
import android.app.IActivityManager;
import android.content.Context;
import android.content.DialogInterface;
import android.os.AsyncTask;

import com.android.settings.R;

public class SystemUtils {

    public static void showSystemUiRestartDialog(final Context context) {
        new AlertDialog.Builder(context)
                .setTitle(R.string.systemui_restart_title)
                .setMessage(R.string.systemui_restart_message)
                .setPositiveButton(R.string.systemui_restart_yes, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        new AsyncTask<Void, Void, Void>() {
                            @Override
                            protected Void doInBackground(Void... params) {
                                try {
                                    ActivityManager am =
                                            (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
                                    IActivityManager ams = ActivityManager.getService();
                                    for (ActivityManager.RunningAppProcessInfo app : am.getRunningAppProcesses()) {
                                        if ("com.android.systemui".equals(app.processName)) {
                                            ams.killApplicationProcess(app.processName, app.uid);
                                            break;
                                        }
                                    }
                                } catch (Exception e) {
                                    e.printStackTrace();
                                }
                                return null;
                            }
                        }.execute();
                    }
                })
                .setNegativeButton(R.string.systemui_restart_not_now, null)
                .show();
    }
}

File: src/com/sakura/settings/utils/SystemUtils.kt
/*
 * Copyright (C) 2025 crDroid Android Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.sakura.settings.utils

import android.content.Context
import android.os.Handler
import android.os.Looper
import android.widget.Toast

import androidx.appcompat.app.AlertDialog

import com.android.settings.R
import com.android.internal.util.sakura.Utils

object SystemUtils {

    @JvmStatic
    fun showSystemUiRestartDialog(context: Context) {
        AlertDialog.Builder(context)
            .setTitle(R.string.systemui_restart_title)
            .setMessage(R.string.systemui_restart_message)
            .setPositiveButton(R.string.systemui_restart_yes) { _, _ ->
                restartSystemUI(context)
            }
            .setNegativeButton(R.string.systemui_restart_not_now, null)
            .show()
    }

    @JvmStatic
    fun restartSystemUI(context: Context) {
        Toast.makeText(
            context,
            R.string.systemui_restart_process,
            Toast.LENGTH_LONG
        ).show()

        Handler(Looper.getMainLooper()).postDelayed({
            Utils.restartSystemUI()
        }, 2000) // 2-second delay
    }
} 